### PR TITLE
Attempt remove net472 as a targets for packages

### PR DESCRIPTION
**Pull request by AndreyTretyak**
Theoretically, we don't need net472 as a target anymore since it could use netstandard2.0 instead, we'll see how it will work :)

### CODE REVIEW

**Review by muiriswoulfe — APPROVED** [review]

**Comment by muiriswoulfe** [comment]
If you search the repo for "net472", you can see some other items related to this framework -- references in .props files and a very small code workaround to facilitate use of the framework. Both of these should be possible to remove now with this change.

**Comment by AndreyTretyak** [comment]
> If you search the repo for "net472", you can see some other items related to this framework -- references in .props files and a very small code workaround to facilitate use of the framework. Both of these should be possible to remove now with this change.

Thank you for noticing that. Unfortunately, some of them are needed but was able to unify some of our targets

File: Dependencies.props [hunk 1]
```diff
@@ -5,25 +5,24 @@
...
-    <MSTest_TestAdapter_Version>2.1.1</MSTest_TestAdapter_Version>
-    <MSTest_TestFramework_Version>2.1.1</MSTest_TestFramework_Version>
+    <MSTest_TestAdapter_Version>2.1.2</MSTest_TestAdapter_Version>
+    <MSTest_TestFramework_Version>2.1.2</MSTest_TestFramework_Version>
     <System_Configuration_ConfigurationManager_Version>4.7.0</System_Configuration_ConfigurationManager_Version>
     <System_ServiceModel_Primitives_Version>4.7.0</System_ServiceModel_Primitives_Version>
     <System_Threading_Tasks_Version>4.3.0</System_Threading_Tasks_Version>
-    <Xunit_Runner_VisualStudio_Version>2.4.2</Xunit_Runner_VisualStudio_Version>
+    <Xunit_Runner_VisualStudio_Version>2.4.3</Xunit_Runner_VisualStudio_Version>
+    <Moq_Version>4.14.5</Moq_Version>
```
Comment (muiriswoulfe) [inline Dependencies.props:17]: Consider sorting. I believe the list is otherwise sorted.

Comment (AndreyTretyak) [reply Dependencies.props]: Oh, missed that, thank you, sorted now

File: src/Extensions/Hosting.Services.Web/Microsoft.Omex.Extensions.Hosting.Services.Web.csproj [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreVersion)</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreVersions)</TargetFrameworks>
```
Comment (muiriswoulfe) [inline src/Extensions/Hosting.Services.Web/Microsoft.Omex.Extensions.Hosting.Services.Web.csproj:3]: Just to confirm -- is this not available for .NET Standard by design?

File: Directory.Build.props [hunk 3]
```diff
@@ -3,11 +3,14 @@
...
     <NetStandardVersions>netstandard2.0</NetStandardVersions>
-    <NetFrameworkVersions>net472</NetFrameworkVersions> <!-- Framework version that should be targeted alongside with .net standard -->
-    <LibraryTargetFrameworks>$(NetCoreVersions);$(NetStandardVersions);$(NetFrameworkVersions)</LibraryTargetFrameworks>
-    <LibraryTargetFrameworks Condition="'$(OS)' != 'Windows_NT'">$(NetCoreVersions);$(NetStandardVersions)</LibraryTargetFrameworks>
+    <LibraryTargetFrameworks>$(NetCoreVersions);$(NetStandardVersions)</LibraryTargetFrameworks>
     <ExecutableTargetFrameworks>$(NetCoreVersions)</ExecutableTargetFrameworks>
-</PropertyGroup>
+  </PropertyGroup>
+  <PropertyGroup Label="UnitTest Targets">
+    <NetFameworkVersion>net472</NetFameworkVersion>
```
Comment (muiriswoulfe) [inline Directory.Build.props:10]: Do we wish to retain .NET 4.7.2 here?

File: src/Extensions/Hosting.Services.Web/Microsoft.Omex.Extensions.Hosting.Services.Web.csproj (hunk 2 above)
Comment (AndreyTretyak) [reply src/Extensions/Hosting.Services.Web/Microsoft.Omex.Extensions.Hosting.Services.Web.csproj:3]: Yes, Asp .Net Core only supports net core starting from version 3

File: Directory.Build.props [hunk 4]
```diff
@@ -3,11 +3,14 @@
...
+    <UnitTestTargetValues>$(NetCoreVersions);$(NetFameworkVersion)</UnitTestTargetValues>
```
Comment (muiriswoulfe) [inline Directory.Build.props:11]: Given that the non-unit test version of this is `LibraryTargetFrameworks`, I think `UnitTestTargetFrameworks` might be a better name here

File: tests/Extensions/Directory.Build.props [hunk 5]
```diff
@@ -1,19 +1,14 @@
 <Project>
   <Import Project="$([MSBuild]::GetPathOfFileAbove('$(_DirectoryBuildPropsFile)', '$(MSBuildThisFileDirectory)../'))" />
   <PropertyGroup>
-    <NetFameworkVersion>net472</NetFameworkVersion>
-    <NetCoreVersion>netcoreapp3.1</NetCoreVersion>
-    <TargetFrameworksValue>$(NetFameworkVersion);$(NetCoreVersion)</TargetFrameworksValue>
     <LangVersion>latest</LangVersion>
     <Nullable>enable</Nullable>
-    <IsPackable>false</IsPackable>
-    <IsTestProject>true</IsTestProject>
```
Comment (muiriswoulfe) [inline tests/Extensions/Directory.Build.props:10]: Do we definitely want to remove `IsTestProject`? It seems like something that would be beneficial, but I can't find much useful information about it on the web

**Review by muiriswoulfe — APPROVED** [review]
A few small comments

File: tests/Extensions/Directory.Build.props (hunk 5 above)
Comment (AndreyTretyak) [reply tests/Extensions/Directory.Build.props:10]: It's defined in the `tests/Directory.Build.props` that is also applied here

File: Directory.Build.props (hunk 3 above)
[the remark this answers is not in the document]
Comment (AndreyTretyak) [reply Directory.Build.props:10]: Thank you missed that one

File: Directory.Build.props (hunk 4 above)
Comment (AndreyTretyak) [reply Directory.Build.props]: Makes sense, thank you

**Review by muiriswoulfe — APPROVED** [review]

**Review by gikoning — APPROVED** [review]

**Pull request merged**